ipa-core/src/protocol/context/dzkp_validator.rs

```diff
@@ -559,36 +515,78 @@ impl Batch {
             .values()
             .flat_map(MultiplicationInputsBatch::get_field_values_from_left_prover::<DF>)
     }
-}
 
-/// Corresponds to `MaliciousAccumulator` of the MAC based malicious validator.
-#[derive(Clone, Debug)]
-pub struct DZKPBatch {
-    inner: Weak<Mutex<Batch>>,
-}
-
-impl DZKPBatch {
-    /// pushes values of a record, i.e. segment, to a `Batch`
-    ///
     /// ## Panics
-    /// Panics when mutex is poisoned or `segments` have different lengths within `gate`
-    pub fn push(&self, gate: Gate, record_id: RecordId, segment: Segment) {
-        let arc_mutex = self.inner.upgrade().unwrap();
-        // LOCK BEGIN
-        let mut batch = arc_mutex.lock().unwrap();
-        batch.push(gate, record_id, segment);
-        // LOCK END
-    }
+    /// If `usize` to `u128` conversion fails.
+    pub(super) async fn validate(self, ctx: Base<'_>) -> Result<(), Error> {
```

Comment:
This used to be `MaliciousDZKPValidator::validate_chunk`. Now it's a method on the batch.